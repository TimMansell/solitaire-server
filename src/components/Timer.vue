<template>
  <div>
    Time:
    <span data-test="timer">{{ timer }}</span
    >s
  </div>
</template>

<script>
import { mapGetters, mapActions } from 'vuex';

export default {
  name: 'Stats',
  computed: {
    ...mapGetters(['timer', 'isGamePaused']),
  },
  mounted() {
    this.setGameTimer();
  },
  watch: {
    isGamePaused(val, oldVal) {
      if (val.isPaused === oldVal.isPaused) return;

      if (val.isPaused) {
        this.clearTimer();
      } else {
        this.setGameTimer();
      }
    },
  },
  methods: {
    ...mapActions(['updateTimer']),
    initGameTimer() {
      return window.setInterval(() => this.updateTimer(), 1000);
    },
    setGameTimer() {
      const { isPaused } = this.isGamePaused;

      if (!isPaused) {
        this.gameTimer = this.initGameTimer();
      }
    },
    clearTimer() {
      clearInterval(this.gameTimer);
    },
  },
};
</script>
