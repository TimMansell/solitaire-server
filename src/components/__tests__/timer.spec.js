import { shallowMount } from '@vue/test-utils';
import Timer from '@/components/Timer.vue';

describe('Timer.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(Timer, {
      computed: {
        timer: () => 1,
        isGamePaused: () => ({
          paused: true,
        }),
      },
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should show 10 seconds on the timer', () => {
    const wrapper = shallowMount(Timer, {
      computed: {
        timer: () => 10,
        isGamePaused: () => ({
          paused: true,
        }),
      },
    });

    expect(wrapper.find('[data-test="timer"]').text()).toContain('1');
  });
});
