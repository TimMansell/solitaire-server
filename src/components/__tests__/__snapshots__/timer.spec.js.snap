// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer.vue matches snapshot 1`] = `
<div>
  Time:
  <span data-test="timer">1</span>s
</div>
`;
